src/OpenSSL/SSL.py

```diff
@@ -701,6 +711,35 @@ def set_default_verify_paths(self):
         """
         set_result = _lib.SSL_CTX_set_default_verify_paths(self._context)
         _openssl_assert(set_result == 1)
+        num = self._check_num_store_objects()
+        if num == 0:
+            self._fallback_default_verify_paths()
+
+    def _check_num_store_objects(self):
+        """
+        Checks how many objects are present in the Context's X509Store.
+
+        :return: int
+        """
+        store = self.get_cert_store()
+        sk_obj = _lib.X509_STORE_get0_objects(store._store)
+        _openssl_assert(sk_obj != _ffi.NULL)
+        return _lib.sk_X509_OBJECT_num(sk_obj)
+
+    def _fallback_default_verify_paths(self):
+        """
+        Default verify paths are based on the compiled version of OpenSSL.
+        However, when pyca/cryptography is compiled as a manylinux1 wheel
+        that compiled location can potentially be wrong. So, like Go, we
+        will try a predefined set of paths and attempt to load roots
+        from there.
+
+        :return: None
+        """
+        for cafile in _CERTIFICATE_FILE_LOCATIONS:
+            if os.path.isfile(cafile):
+                self.load_verify_locations(cafile)
```

Comment:
If the file exists but is invalid this will currently raise an exception. Do we want to catch this and silently swallow it?